src/AmqpBundle/Amqp/DataCollector.php

```diff
@@ -101,4 +101,12 @@ public function getAvgExecutionTime()
     {
         return ($this->getTotalExecutionTime()) ? ($this->getTotalExecutionTime() / count($this->data['commands'])) : 0;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reset()
+    {
+        $this->data = [];
```

Comment:
I'd rather do:
```php
$this->data['commands'] = [];
```

(because this is what is done in `__construct()`)

but maybe I misunderstood the "reset" purpose?

Reply:
It should be the entire data, not just commands.

Reply:
this DataCollector seems to require the `commands` key for `data` array

else we'll have some issues with other methods (i.e. `getCommands`, `getTotalExecutionTime` and `getAvgExecutionTime`)

Reply:
I'll have a closer look at it again today to make sure this works as intended.